src/cpp/rtps/messages/MessageReceiver.cpp

```diff
@@ -958,13 +930,10 @@ bool MessageReceiver::proc_Submsg_InfoDST(CDRMessage_t* msg,SubmessageHeader_t*
         this->destGuidPrefix = guidP;
         logInfo(RTPS_MSG_IN,IDSTRING"DST RTPSParticipant is now: "<< this->destGuidPrefix);
     }
-    //Is the final message?
-    if(smh->submessageLength == 0)
-        *last = true;
     return true;
 }
 
-bool MessageReceiver::proc_Submsg_InfoSRC(CDRMessage_t* msg,SubmessageHeader_t* smh, bool* last)
+bool MessageReceiver::proc_Submsg_InfoSRC(CDRMessage_t* msg,SubmessageHeader_t* smh)
```

Comment:
Instead of `if(smh->submessageLength == 20 || smh->submessageLength==0)`. Can we use `smh->submsgLengthLarger`?

Reply:
I have removed submsgLengthLarger and turned submessageLength into a uint32_t, also adding the is_last flag to SubmessageHeader_t